R/facet-wrap.R

```diff
@@ -99,12 +99,17 @@ facet_wrap <- function(facets, nrow = NULL, ncol = NULL, scales = "fixed",
     strip.position <- if (switch == "x") "bottom" else "left"
   }
   strip.position <- arg_match0(strip.position, c("top", "bottom", "left", "right"))
-  if (!(is.null(ncol) || (is_integerish(ncol, 1) && ncol > 0))) {
-    cli::cli_abort("{.arg ncol} must be a positive scalar integer or {.val NULL}")
-  }
-  if (!(is.null(nrow) || (is_integerish(nrow, 1) && nrow > 0))) {
-    cli::cli_abort("{.arg nrow} must be a positive scalar integer or {.val NULL}")
-  }
+
+  check_number_whole(ncol, allow_null = TRUE, min = 1)
+  check_number_whole(nrow, allow_null = TRUE, min = 1)
+
+  # if (!(is.null(ncol) || (is_integerish(ncol, 1) && ncol > 0))) {
+  #   cli::cli_abort("{.arg ncol} must be a positive scalar integer or {.val NULL}")
+  # }
+  # if (!(is.null(nrow) || (is_integerish(nrow, 1) && nrow > 0))) {
+  #   cli::cli_abort("{.arg nrow} must be a positive scalar integer or {.val NULL}")
+  # }
```

Comment:
Currently gives misleading message.
I've filed https://github.com/r-lib/rlang/issues/1591 in the hope to improve the message a bit.
If we can't improve the message, we'll revert back to the old check.

Reply:
Fixed by https://github.com/r-lib/rlang/pull/1594